Code/Engine/ECS/Components/TransformComponent.h

```diff
@@ -12,6 +13,59 @@ namespace sad::ecs
 	*/
 	struct TransformComponent
 	{
-		Transform* m_Transform;
+		core::Pointer<Transform> m_Transform;
 	};
+
+	inline void to_json(nlohmann::json& j, const sad::ecs::TransformComponent& p)

```

Comment:
Same comment with above - small change to variable names would be sick (lines 19 and 55).